src/app/lobby/playersPanel.tsx

```diff
@@ -0,0 +1,72 @@
+import { Player } from "@/api/session";
+import { PlayerDirection } from "@/game_engine/gameModels";
+import PlayerPanel from "./playerPanel";
+import { useForceSwap, usePromoteHost } from "@/api/session/lobby";
+import { useCallback, useEffect, useState } from "react";
+
+export default function PlayersPanel({
+  players,
+  userId,
+  host,
+}: {
+  players: Player[];
+  userId: string;
+  host: Player | undefined;
+}) {
+  const promoteHost = usePromoteHost();
+  const forceSwap = useForceSwap();
+  const [positionsToSwap, setPositionsToSwap] = useState<PlayerDirection[]>([]);
+
+  const handleForceSwap = useCallback(() => {
+    if (forceSwap.loading || positionsToSwap.length != 2) return;
+    forceSwap.trigger({ first: positionsToSwap[0], second: positionsToSwap[1] });
+  }, [forceSwap, positionsToSwap]);
+
+  const addPositionToSwap = useCallback((position: PlayerDirection) => {
+    setPositionsToSwap((positions) => {
+      const newPositions = [...positions];
+      if (newPositions.length < 2) newPositions.push(position);
+      return newPositions;
+    });
+  }, []);
+
+  const handleSetHost = useCallback((userId: string) => {
+    if (promoteHost.loading) return;
+    promoteHost.trigger({ userId: userId });
+  }, [promoteHost]);
+
+
+  useEffect(() => {
+    if (positionsToSwap.length == 2) {
+      if (positionsToSwap[0] != positionsToSwap[1]) handleForceSwap();
+      setPositionsToSwap([]);
+    }
+  }, [positionsToSwap, handleForceSwap]);
+
+  function directionToPlayer(direction: PlayerDirection, key: number) {
+    const player = players.find((p) => p.position == direction);
+    return (
+      <PlayerPanel
+        key={key}
+        player={players.find((p) => p.position == direction)}
+        userId={userId}
+        host={host}
+        position={direction}
+        addPositionToSwap={addPositionToSwap}
+        positionsToSwap={positionsToSwap}
+        promoteHost={handleSetHost}
+      />
+    );
+  }
+
```

Comment:
to musi byc callback bo nie aktualizacuja mu sie zmienne jak np. host, handleSetHost itp